modules/test/playwright/tests/portal-security-content-security-policy/contentSecurityPolicy.spec.ts

```diff
@@ -5,40 +5,822 @@
+import {performLoginViaApi} from '../../utils/performLogin';
+import getFragmentDefinition from '../layout-content-page-editor-web/utils/getFragmentDefinition';
+import getPageDefinition from '../layout-content-page-editor-web/utils/getPageDefinition';
 
 export const test = mergeTests(
+	apiHelpersTest,
 	contentSecurityPolicyPagesTest,
-	featureFlagsTest({
-		'LPS-134060': {enabled: true},
-	}),
-	loginTest()
+	isolatedSiteTest,
+	loginTest(),
+	pageEditorPagesTest,
+	virtualInstancesPagesTest
 );
 
-test('Content Security Policy Smoke Test', async ({
+test.afterEach(
+	'Reset CSP configuration',
+	async ({contentSecurityPolicyPage, page}) => {
+		await page.goto('/');
+
+		if (await page.getByRole('button', {name: 'Sign In'}).isVisible()) {
+			await performLoginViaApi(page, 'test');
+		}
+
+		await contentSecurityPolicyPage.goto();
+
+		await contentSecurityPolicyPage.resetCSPConfiguration();
+	}
+);
+
+test("CSP connect-src allows connections to 'self'", async ({
+	apiHelpers,
+	contentSecurityPolicyPage,
+	page,
+	pageEditorPage,
+	site,
+}) => {
+	await contentSecurityPolicyPage.gotoAndConfigurePolicy(
+		`connect-src 'self'`
+	);
+
+	const layout = await apiHelpers.headlessDelivery.createSitePage({
+		pageDefinition: getPageDefinition([
+			getFragmentDefinition({
+				id: getRandomString(),
+				key: 'BASIC_COMPONENT-html',
+			}),
+		]),
+		siteId: site.id,
+		title: getRandomString(),
+	});
+
+	await pageEditorPage.goto(layout, site.friendlyUrlPath);
+
+	const portalURL = liferayConfig.environment.baseUrl;
+
+	await pageEditorPage.editHTMLEditable({
+		editableId: 'element-html',
+		fragmentId: await pageEditorPage.getFragmentId('HTML'),
+		value: `<a  ping="${portalURL}" href="${portalURL}" target="_blank">
+		 			Test CSP
+					<script>
+						const response = fetch("${portalURL}");
+
+						const xmlHttpRequest = new XMLHttpRequest();
+						xmlHttpRequest.open("GET", "${portalURL}");
+						xmlHttpRequest.send();
+
+						const webSocket = new WebSocket("${portalURL}");
+
+						const eventSource = new EventSource("${portalURL}");
+
+						navigator.sendBeacon("${portalURL}", {
+						/* … */
+						});
+					</script>
+				</a>`,
+	});
+
+	await pageEditorPage.publishPage();
+
+	const cspErrors = [];
+
+	page.on('console', (msg) => {
+		if (
+			msg.type() === 'error' &&
+			msg
+				.text()
+				.includes('Content Security Policy directive: "connect-src')
+		) {
+			cspErrors.push({text: msg.text(), type: msg.type()});
+		}
+	});
+
+	page.on('requestfailed', (request) => {
+		if (
+			request.url().includes('localhost') &&
+			request.failure().errorText.includes('csp')
+		) {
+			cspErrors.push({
+				failure: request.failure(),
+				url: request.url(),
+			});
+		}
+	});
+
+	await page.goto(`/web/${site.name}/${layout.friendlyUrlPath}`);
+
+	await page.getByRole('link', {name: 'Test CSP'}).click({
+		button: 'middle',
+	});
+
+	expect(cspErrors.length).toBeLessThanOrEqual(0);
```

Comment:
These expect statements aren't all uniform.  Please adjust to the following:
`expect(cspErrors).toHaveLength(x)`

There are some expecting `>=` or `<=` a value other than `0`, these should have a specific expected number instead.

Reply:
Also, sometimes the variable is `cspErrors`, and other times it is just `errors`.  Please make them all `cspErrors` if that name makes more sense.